torch/serialization.py

```diff
@@ -90,7 +90,34 @@ def _is_zipfile(f) -> bool:
     return read_bytes == local_header_magic_number
 
 
-def register_package(priority, tagger, deserializer):
+def register_package(
+    priority: int,
+    tagger: Callable[[Union[Storage, torch.storage.TypedStorage, torch.storage.UntypedStorage]], Optional[str]],
+    deserializer: Callable[[Union[Storage, torch.storage.TypedStorage, torch.storage.UntypedStorage], str],
+                           Optional[Union[Storage, torch.storage.TypedStorage, torch.storage.UntypedStorage]]]):
+    '''
+    Registers callables for tagging and deserializing storage objects with an associated priority.
+    Tagging associates a device with a storage object at save time while deserializing moves a
+    storage object to an appropriate device at load time. :attr:`tagger` and :attr:`deserializer`
+    are run in the order given by their :attr:`priority` until a tagger/deserializer returns a
+    value that is not `None`.
+
+    To override the deserialization behavior for a device in the global registry, one can register a
+    tagger with a higher priority than the existing tagger.
+
+    This function can also be used to register a tagger and deserializer for new devices.
+
+    Args:
+        priority: Indicates the priority associated with the tagger and deserializer, where a lower
+            value indicates higher priority.
+        tagger: Callable that takes in a storage object and returns its tagged device as a string
```

Comment:
Just curious, why can these callables return None? What's the use case when that happens?

The docs are pretty clear already, but an example would help solidify understanding.

Reply:
If a given tagger/deserializer returns None, we continue to the next deserializer until something returns something non-None at which point we terminate and return. 

Tried to capture this here, but not sure if it's very clear :b

> Taggers/deserializers are run in the order given by their :attr:`priority` until a tagger/deserializer returns a
    value that is not `None`.